src/builder/commands/generic.rs

```diff
@@ -303,19 +300,18 @@ impl BuildStep for RunAs {
         for i in self.supplementary_gids.iter() {
             hash.text("supplementary_gids", &i);
         }
-        hash.field("RunAs", &self.script);
+        hash.field("script", &self.script);
         Ok(())
     }
     fn build(&self, guard: &mut Guard, build: bool)
         -> Result<(), StepError>
     {
         if build {
-            let mut cmd = try!(build_command(&mut guard.ctx,
-                                             &["/bin/sh".to_string(),
-                                               "-exc".to_string(),
-                                               self.script.to_string()],
-                                             &Path::new("/work"),
-                                             &[]));
+            let mut cmd = Command::new(&"/bin/sh".to_string());
+            setup_command(&guard.ctx, &mut cmd);
+            cmd.args(&["-exc".to_string(), self.script.to_string()]);
```

Comment:
This is similar, you don't need `to_string()` conversion, which is allocating a `String`. and you don't need to create a slice, just:
```
cmd.arg("-exc");
cmd.arg(&self.script)
```

(The slice is cost-less, but using separate call allows to use different type for each argument, if needed)